packages/@aws-cdk/pipelines/lib/codepipeline/codepipeline-source.ts

```diff
@@ -291,6 +291,18 @@ export interface S3SourceOptions {
    * @default - The bucket name
    */
   readonly actionName?: string;
+
+  /**
+   * The Role in which context's this Action will be executing in.
+   * The Pipeline's Role will assume this Role
+   * (the required permissions for that will be granted automatically)
+   * right before executing this Action.
+   * This Action will be passed into your {@link IAction.bind}
+   * method in the {@link ActionBindOptions.role} property.
```

Comment:
```suggestion
   * The role that will be assumed by the pipeline prior to executing
   * the `S3Source` action.
```

Feel free to expand a tad further if you want, but in general, there's no need to go into the implementation detail in the documentation of a property. We also do this `role` property all over the CDK, so generally people should know the drill.